cargo-dist/src/platform.rs

```diff
@@ -0,0 +1,414 @@
+            (
+                format!("{system}musl-static{abigunk}"),
+                format!("{degunked_target}-static"),
+            )
+        } else {
+            (target.to_owned(), degunked_target.to_owned())
+        };
+
+        // First, add the target itself as a HostNative entry
+        res.push((
+            target.clone(),
+            PlatformEntry {
+                quality: SupportQuality::HostNative,
+                runtime_conditions: archive.native_runtime_conditions.clone(),
+                archive_idx,
+            },
+        ));
+
+        // If this is a static linux libc, say it can support any linux at ImperfectNative quality
+        for &static_libc in LINUX_STATIC_LIBCS {
+            let Some(system) = degunked_target.strip_suffix(static_libc) else {
+                continue;
+            };
+            for &libc in LINUX_STATIC_REPLACEABLE_LIBCS {
+                res.push((
+                    format!("{system}{libc}{abigunk}"),
+                    PlatformEntry {
+                        quality: SupportQuality::ImperfectNative,
+                        runtime_conditions: archive.native_runtime_conditions.clone(),
+                        archive_idx,
+                    },
+                ));
+            }
+            break;
+        }
+
+        // universal2 macos binaries are totally native for both arches, but bulkier than
+        // necessary if we have builds for the individual platforms too.
+        if target == TARGET_MACOS_UNIVERSAL2 {
+            res.push((
+                TARGET_X64_MAC.to_owned(),
+                PlatformEntry {
+                    quality: SupportQuality::BulkyNative,
+                    runtime_conditions: archive.native_runtime_conditions.clone(),
+                    archive_idx,
+                },
+            ));
+            res.push((
+                TARGET_ARM64_MAC.to_owned(),
+                PlatformEntry {
+                    quality: SupportQuality::BulkyNative,
+                    runtime_conditions: archive.native_runtime_conditions.clone(),
+                    archive_idx,
+                },
+            ));
+        }
+
+        // x86_32 macos binaries ran fine on x86_64, but it's Imperfect compared to actual x86_64 binaries
+        // ...UP UNTIL macOS Catalina (macOS 10.15, October 2019)!
+        //
+        // FIXME: add some condition for "check the macos version" if we care about this!
+        if target == TARGET_X86_MAC {
+            res.push((
+                TARGET_X64_MAC.to_owned(),
+                PlatformEntry {
+                    quality: SupportQuality::ImperfectNative,
+                    runtime_conditions: archive.native_runtime_conditions.clone(),
+                    archive_idx,
+                },
+            ));
+        }
+
+        // If this is x64 macos, say it can support arm64 macos using Rosetta2
```

Comment:
Minor nitpick for clarity: I initially read this as meaning that x86_64 Macs could run arm64, rather than the other way around.